### ISSUE TITLE
Implement training loop for geodetic loss

### ISSUE THREAD

**Issue by albangossard** [ODINN-SciML/MassBalanceMachine#112]
Since we moved to a very specific use case where the batches are not independent from each other, we can no longer rely on `scikit-learn` tools. We need to implement our own training loop in the case of the neural network (with pytorch).

What have to be implemented in this task:
- the validation function
- the training function for one iteration (a subset of glaciers)
- the function to save and load a model
- the training loop that performs the training for a given number of epochs, calls the training and validation functions and keep track of the training histories.

In order to keep track of the convergence histories, we can use both tensorboard (for visualization it is very convenient), and save the histories in a dedicated file that is convenient to use for post-processing (json or any format that has more compression for example?).
